Doc reorg: Configuring CumulusCI (#2117)
SFDO-Tooling/CumulusCI: docs/env_var_reference.rst

> Environment Variables

Reviewer: ```suggestion
Set Environment Variables
```

=====================

> CumulusCI has environment variables that are useful when CumulusCI is being run inside of web applications such as MetaCI, MetaDeploy, and Metecho.

Reviewer: ```suggestion
CumulusCI environment variables are useful when CumulusCI is run inside of web applications such as MetaCI, MetaDeploy, and Metecho.
```

> The following is a reference list of available environment variables that can be set.

Reviewer: ```suggestion
Here's a list of environment variables that can be configured.
```

``CUMULUSCI_AUTO_DETECT`` 

>     Set this environment variable to autodetect branch and commit information from ``HEROKU_TEST_RUN_BRANCH`` and ``HEROKU_TEST_RUN_COMMIT_VERSION`` environment variables.

Reviewer 2: Might be worth adding "This is relevant when running CumulusCI inside Heroku CI, and is not commonly used."

>     Set this environment variable to autodetect branch and commit information from ``HEROKU_TEST_RUN_BRANCH`` and ``HEROKU_TEST_RUN_COMMIT_VERSION`` environment variables.

Reviewer: ```suggestion
    Autodetect branch and commit information from ``HEROKU_TEST_RUN_BRANCH`` and ``HEROKU_TEST_RUN_COMMIT_VERSION`` environment variables.
```

``CUMULUSCI_DISABLE_REFRESH``

>     If present, will instruct CumulusCI to not refresh oAuth tokens for orgs.

Reviewer 3: ```suggestion
    If present, will instruct CumulusCI to not refresh OAuth tokens for orgs.
```

>     If present, will instruct CumulusCI to not refresh OAuth tokens for orgs.

Reviewer: ```suggestion
    If present, prevent refresh of OAuth tokens for orgs.
```

``CUMULUSCI_KEY``

    An alphanumeric string used to encrypt org credentials at rest when an OS keychain is not available.

``CUMULUSCI_REPO_URL``
    

>     Used for specifying a Github Repository for CumulusCI to use when running in a CI environment.

Reviewer: ```suggestion
    Specify a Github repository for CumulusCI to use when running in a CI environment.
```

``GITHUB_APP_ID``

    Your GitHub App's identifier.

``GITHUB_APP_KEY``

>     Contents of a JSON Web Token (JWT) used to authenticate a GitHub app (see `GitHub’s docs <https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app>`_).

Reviewer: ```suggestion
    Contents of a JSON Web Token (JWT) used to authenticate a GitHub app (see `Authenticating as a GitHub App <https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app>`_) in GitHub documentation.
```
For accessibility reasons, make the hyperlinked text match the title of the thing that we're linking to.

``GITHUB_TOKEN``

    A GitHub `personal access token <https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line>`_.

> 

Reviewer 2: Probably should explain that _either_ GITHUB_TOKEN or the combo of GITHUB_APP_ID + GITHUB_APP_KEY can be used to override how CumulusCI authenticates to github (instead of using the github service from the keychain).

``HEROKU_TEST_RUN_BRANCH``

>     Used for specifying a specific branch to test against in a Heroku CI environment

Reviewer: ```suggestion
    Specify a specific branch to test against in a Heroku CI environment.
```

``HEROKU_TEST_RUN_COMMIT_VERSION``

>     Used to specify a specific commit to test against in a Heroku CI environment.

Reviewer: ```suggestion
    Specify a specific commit to test against in a Heroku CI environment.
```

``SFDX_CLIENT_ID``

>     Client ID for a Connected App used to authenticate to a persistent org, e.g. a Developer Hub. Set with SFDX_HUB_KEY.

Reviewer: ```suggestion
    Client ID for a connected app used to authenticate to a persistent org, such as a Developer Hub. Set with SFDX_HUB_KEY.
```
https://confluence.internal.salesforce.com/display/DOCTEAM/Latin+expressions

``SFDX_HUB_KEY``
    
    Contents of JSON Web Token (JWT) used to authenticate to a persistent org, e.g. a Developer Hub.  Set with SFDX_CLIENT_ID.

> 

Reviewer 2: We should note that when both SFDX_CLIENT_ID and SFDX_HUB_KEY are set, CumulusCI will use the JWT oauth flow to refresh org access tokens instead of the refresh token flow.
